### PR TITLE
Preparations of deep tau ID for hlt phase2 (backport to CMSSW_11_1_X)

### PR DESCRIPTION
**Pull request by swozniewski**
#### PR description:

Backports #30761 #31744 #32676 #32838 in order to have DeepTauID developments available for phase2 HLT TDR

Prepare DeepTauID such that it can run on AOD inputs and be used within HLT sequences.
#30761 was some central clean up that interferes with the other two PRs. For convenience and consistent state of the code it is included.

#### PR validation:
- code checks, format passed
- unit tests passed
- limited matrix tests passed

### CODE REVIEW

**Comment by fwyzard** [comment]
please test

File: RecoTauTag/RecoTau/src/PositionAtECalEntranceComputer.cc [hunk 1]
```diff
@@ -16,7 +16,8 @@ void PositionAtECalEntranceComputer::beginEvent(const edm::EventSetup& es) {
   bField_z_ = bField->inTesla(GlobalPoint(0., 0., 0.)).z();
 }
 
-reco::Candidate::Point PositionAtECalEntranceComputer::operator()(const reco::Candidate* particle, bool& success) const {
+reco::Candidate::Point PositionAtECalEntranceComputer::operator()(const reco::Candidate* particle,
+                                                                  bool& success) const {
```
Comment (slava77) [inline RecoTauTag/RecoTau/src/PositionAtECalEntranceComputer.cc:20]: this is not a part of the listed PRs used for backport.
Apparently, the change is coming from 
https://github.com/cms-sw/cmssw/pull/30571/files#diff-225356b5230d327db61da9f1a9a6514e03ec42126c806f2a46fc54c97a7a64ebL19-R20

it seems acceptable to include though.

Comment (swozniewski) [reply RecoTauTag/RecoTau/src/PositionAtECalEntranceComputer.cc:20]: I think it is rather this commit https://github.com/cms-sw/cmssw/commit/9457485ec9f806bc1f877d8a28bbbc2d8d87f50f which is probably not backported but where I applied the code-format at some point. Sorry for the confusion!

**Comment by perrotta** [comment]
please test

**Comment by fwyzard** [comment]
+hlt

The only HLT specific change is the addition of a customisation function.

**Comment by fwyzard** [comment]
@slava77 @perrotta is this good for RECO ?

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/32831	6834218
- the backport is appropriate under a full no-change policy (no changes in the default setup are expected)
    - the diffs are in agreement with the PR description (#30761 -> #31744 -> #32676 -> #32838 )
    - comparing 11_1_X -> master in the modified files, except for RecoTauTag/RecoTau/python/tools/runTauIdMVA.py, there are technical modifications from #30364 and #30805  which should be OK to skip, and #32614 is 113X-specific and does not apply
    - runTauIdMVA.py in this PR has just an addition of `is_online` parameter, and it seems OK to skip other changes in this file as unrelated to deepTau
- jenkins tests pass and comparisons with the baseline show no (relevant) differences

**Comment by qliphy** [comment]
+1

**Pull request merged**